Cargo.toml

```diff
@@ -19,6 +19,7 @@ hkdf = "0.11"
 log = "0.4"
 rand_core = "0.6"
 rand = "0.8"
+rug = "1.15.0"
```

Comment:
this dependency uses gmpz under the hood, but includes the gmp source code along with the rust bindings, so this is all that is needed to use it.

according to docs, "[Rug also depends on the GMP, MPFR and MPC libraries through the low-level FFI bindings in the gmp-mpfr-sys crate, which needs some setup to build](https://docs.rs/rug/latest/rug/#using-rug)".

that setup is straightforward, eg on linux you need to have `diffutils`, `gcc`, `m4` and `make` installed. on macos, you just need to run `xcode-select --install`.

Reply:
gmp (including rug) can have a significant perf improvement over the pure-rust [num_bigint](https://docs.rs/num-bigint/latest/num_bigint/).

according to [benchmarks here](https://github.com/vks/discrete-log), which solve for discrete log, gmp can be 6-8x faster than pure rust implementation